### PR TITLE
add check_mode support to ec2_vpc_net module

### PR DESCRIPTION
**Pull request by robbwagoner**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

ec2_vpc_net
##### ANSIBLE VERSION

```
ansible 2.0.2.0
  config file =
  configured module search path = Default w/o overrides
```
##### SUMMARY

Adds check_mode support (`--check`) to ec2_vpc_net module.

_BEFORE_

``` shell
$ ansible-playbook vpc-check.yml -i inventory/local -vvv --check
PLAYBOOK: vpc-check.yml ********************************************************
1 plays in vpc-check.yml

PLAY [VPC Check] ***************************************************************

TASK [ec2_vpc_net] *************************************************************
task path: vpc-check.yml:7
ESTABLISH LOCAL CONNECTION FOR USER: robb
localhost EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python'
skipping: [localhost] => {"changed": false, "invocation": {"module_args": {"cidr_block": "192.168.254.0/24", "dns_hostnames": true, "dns_support": true, "name": "ansible-ec2-vpc-net-c
heck", "region": "us-east-1", "state": "present", "tags": {"Name": "ansible-ec2-vpc-net-check", "owner": "@robbwagoner"}, "tenancy": "default"}, "module_name": "ec2_vpc_net"}, "msg": "
remote module does not support check mode", "skipped": true}

TASK [debug] *******************************************************************
task path: vpc-check.yml:20
ok: [localhost] => {
    "vpc_details": {
        "changed": false,
        "msg": "remote module does not support check mode",
        "skipped": true
    }
}

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0
```

_AFTER_
_for an existing VPC_

``` shell
$ ansible-playbook.sh vpc-check.yml -i inventory/local -vvv --check 
PLAYBOOK: vpc-check.yml ********************************************************
1 plays in vpc-check.yml

PLAY [VPC Check] ***************************************************************

TASK [ec2_vpc_net] *************************************************************
task path: vpc-check.yml:7
ESTABLISH LOCAL CONNECTION FOR USER: robb
localhost EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python'
changed: [localhost] => {"changed": true, "invocation": {"module_args": {"aws_access_key": null, "aws_secret_key": null, "cidr_block": "192.168.254.0/24", "dhcp_opts_id": null, "dns_h
ostnames": true, "dns_support": true, "ec2_url": null, "multi_ok": false, "name": "ansible-ec2-vpc-net-check", "profile": null, "region": "us-east-1", "security_token": null, "state":
 "present", "tags": {"Name": "ansible-ec2-vpc-net-check", "owner": "@robbwagoner"}, "tenancy": "default", "validate_certs": true}, "module_name": "ec2_vpc_net"}, "vpc": {"cidr_block":
 "192.168.254.0/24", "classic_link_enabled": null, "dhcp_options_id": "dopt-1001f575", "id": "vpc-5fb25938", "instance_tenancy": "default", "is_default": false, "state": "available",
"tags": {"Name": "ansible-ec2-vpc-net-check", "owner": "@robbwagoner"}}}

TASK [debug] *******************************************************************
task path: vpc-check.yml:20
ok: [localhost] => {
    "vpc_details": {
        "changed": true,
        "vpc": {
            "cidr_block": "192.168.254.0/24",
            "classic_link_enabled": null,
            "dhcp_options_id": "dopt-1001f575",
            "id": "vpc-6ea15938",
            "instance_tenancy": "default",
            "is_default": false,
            "state": "available",
            "tags": {
                "Name": "ansible-ec2-vpc-net-check",
                "owner": "@robbwagoner"
            }
        }
    }
}

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @robbwagoner for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by resmo** [comment]
LGTM

**Comment by gregdek** [comment]
Thanks again to @robbwagoner for this PR, and thanks @ansible for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by bcoca** [comment]
this seems dupe of #2179 both do somewhat different things, can you coordinate?

**Comment by gregdek** [comment]
Thanks @robbwagoner for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by shawnsi** [comment]
This PR appears to go a bit farther than mine in flushing out check mode support.  It attempts to update the VPC object properties in check mode so a registered variable is up to date.  I'm wondering if this work can be rebased on my change which has been in review and awaiting inclusion for some time.  That way the basic check mode support can go out the door while variable consistency issues are reviewed here.

Thanks for the fixes @robbwagoner.

File: cloud/amazon/ec2_vpc_net.py [hunk 1]
```diff
@@ -144,21 +144,28 @@ def update_vpc_tags(vpc, module, vpc_obj, tags, name):
...
-            return True
-        else:
-            return False
+        if vpc_obj:
+            current_tags = dict((t.name, t.value) for t in vpc.get_all_tags(filters={'resource-id': vpc_obj.id}))
+            if cmp(tags, current_tags):
+                if not module.check_mode:
+                    vpc.create_tags(vpc_obj.id, tags)
+                else:
+                    vpc_obj.tags = tags
```
Comment (shawnsi) [inline cloud/amazon/ec2_vpc_net.py:153]: Maybe I'm missing something but shouldn't this be updated regardless whether check mode is enabled?  The final return values seem to pull from vpc_obj only.

**Comment by shawnsi** [comment]
Does it make sense to continue past https://github.com/robbwagoner/ansible-modules-core/blob/72cb870b7f4333e8b0ac40954f421ed666059b54/cloud/amazon/ec2_vpc_net.py#L242 if check mode is enabled?  Its possible to construct a VPC object still but no VPC ID will be known.  Exiting the module at this stage would simplify downstream logic for managing the `vpc_obj` variable.

**Comment by gregdek** [comment]
Thanks @robbwagoner for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@robbwagoner A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@robbwagoner Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@robbwagoner A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Closing in favor of #2179 to get us moving forward. Apologies @robbwagoner, and sorry for the delay in making a decision here.

**Pull request closed without merging**